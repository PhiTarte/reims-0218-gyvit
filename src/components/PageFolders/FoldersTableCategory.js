import React from "react"
import PropTypes from "prop-types"

const FoldersTableCategory = ({ name = "", direction = null }) => {
  let source, alter
  switch (direction) {
    case "asc":
      source = process.env.PUBLIC_URL.concat("Assets/icon_arrow_down.png")
      alter = "down"
      break
    case "desc":
      source = process.env.PUBLIC_URL.concat("Assets/icon_arrow_up.png")
      alter = "up"
      break
    default:
      source = process.env.PUBLIC_URL.concat("Assets/icon_double_arrow.png")
      alter = "upDown"
      break
  }
  return (
    <div>
      <span>{name + "    "}</span>
      <img src={source} alt={alter} />
    </div>
  )
}

FoldersTableCategory.propTypes = {
  name: PropTypes.string.isRequired,
  direction: PropTypes.string
}

export default FoldersTableCategory
