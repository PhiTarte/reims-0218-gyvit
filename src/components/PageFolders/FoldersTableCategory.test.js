import React from "react"
import ReactDOM from "react-dom"
import renderer from "react-test-renderer"
import FoldersTableCategory from "./FoldersTableCategory"

const name = "Kevin"

it("renders without crashing", () => {
  const div = document.createElement("div")
  ReactDOM.render(<FoldersTableCategory name={name} />, div)
  ReactDOM.unmountComponentAtNode(div)
})

it("renders link html", () => {
  const tree = renderer.create(<FoldersTableCategory name={name} />).toJSON()
  expect(tree).toMatchSnapshot()
})
