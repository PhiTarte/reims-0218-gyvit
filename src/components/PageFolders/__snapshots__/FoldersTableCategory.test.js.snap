// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders link html 1`] = `
<div>
  Kevin
  <img
    alt="upDown"
    src="Assets/icon_double_arrow.png"
  />
</div>
`;
